src/garage/torch/algos/maml.py

```diff
@@ -402,51 +400,47 @@ def evaluate_performance(self, itr, all_samples, loss_before, loss_after,
             tabular.record('KLAfter', kl)
             tabular.record('Entropy', policy_entropy)
 
-        return average_return
+        return rtn
 
 
 class MAMLTrajectoryBatch(
-        collections.namedtuple(
-            'MAMLTrajectoryBatch',
-            ['observations', 'actions', 'rewards', 'valids', 'baselines'])):
+        collections.namedtuple('MAMLTrajectoryBatch', [
+            'paths', 'observations', 'actions', 'rewards', 'valids',
+            'baselines'
+        ])):
     r"""A tuple representing a batch of whole trajectories in MAML.
 
     A :class:`MAMLTrajectoryBatch` represents a batch of whole trajectories
     produced from one environment.
-
     +-----------------------+-------------------------------------------------+
     | Symbol                | Description                                     |
     +=======================+=================================================+
     | :math:`N`             | Trajectory index dimension                      |
     +-----------------------+-------------------------------------------------+
-    | :math:`[T]`           | Variable-length time dimension of each          |
-    |                       | trajectory                                      |
+    | :math:`T`             | Maximum length of a trajectory                  |
     +-----------------------+-------------------------------------------------+
     | :math:`S^*`           | Single-step shape of a time-series tensor       |
     +-----------------------+-------------------------------------------------+
-    | :math:`N \bullet [T]` | A dimension computed by flattening a            |
-    |                       | variable-length time dimension :math:`[T]` into |
-    |                       | a single batch dimension with length            |
-    |                       | :math:`sum_{i \in N} [T]_i`                     |
-    +-----------------------+-------------------------------------------------+
 
     Attributes:
+        paths (list[dict[str, np.ndarray or dict[str, np.ndarray]]]):
```

Comment:
I add `paths` to `MAMLTrajectoryBatch` because I need it to construct `TrajectoryBatch` here https://github.com/rlworkgroup/garage/blob/281ada27351d07817969c9985cd3ab1eb080ea2d/src/garage/torch/algos/maml.py#L386

Reply:
Ah, I see. This is definitely a code smell, but understandable for the time being.
In the future, I think we should change `TrajectoryBatch` to be a type that can be easily subclassed to add additional data (for example, using [`types.SimpleNamespace`](https://docs.python.org/3/library/types.html#types.SimpleNamespace), [`typing.NameTuple`](https://docs.python.org/3/library/typing.html#typing.NamedTuple), or [`dataclass`](https://github.com/ericvsmith/dataclasses).

Reply:
But for this PR I don't think we need to change it, and you can just keep paths.

Reply:
@krzentner if you feel like that's a fully-fleshed idea, please add an issue for it. i'm a little wary of type proliferation but i can see the benefits of subclassing.

Reply:
I've opened an issue  #1226 for this.